crypto/property/property.c

```diff
@@ -898,6 +926,38 @@ int ossl_method_store_cache_get(OSSL_METHOD_STORE *store, OSSL_PROVIDER *prov,
     return res;
 }
 
+int ossl_frozen_method_store_cache_get(OSSL_METHOD_STORE *store,
+    OSSL_PROVIDER *prov, int nid,
+    const char *prop_query, void **method)
+{
+    /*
+     * Query triplet (nid, prov, prop_query) from frozen store with no fallback.
+     */
+
+    /*
+     * TODO: FREEZE: Replace with actual implementation
+     * For now, just store & fetch from global variables
+     */
+    EVP_MD *temp_method;
+
+    if (andrew_method == NULL) {
+        if (!ossl_method_store_cache_get(store, prov, nid, prop_query, (void **)&temp_method))
```

Comment:
how do we ensure that every registered algorithm appears in the cache before it gets frozen?  IIRC the cache only maintains a hashtable of algorithms that have been looked up against a specific provider string.